sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -1483,7 +1483,8 @@ class HistGradientBoostingRegressor(RegressorMixin, BaseHistGradientBoosting):
         than a few hundred samples, it is recommended to lower this value
         since only very shallow trees would be built.
     l2_regularization : float, default=0
-        The L2 regularization parameter. Use ``0`` for no regularization (default).
+        The L2 regularization parameter penalizing large leaf weights. Use ``0``
```

Comment:
That’s not correct. The value of leaves with small hessians are penalized. (And again weight vs value)